pkg/gui/gui.go

```diff
@@ -434,6 +440,54 @@ func (gui *Gui) onUserConfigLoaded() error {
 	return nil
 }
 
+func (gui *Gui) checkForChangedConfigsThatDontAutoReload(oldConfig *config.UserConfig, newConfig *config.UserConfig) error {
+	configsThatDontAutoReload := []string{
+		"Git.AutoFetch",
+		"Git.AutoRefresh",
+		"Refresher.RefreshInterval",
+		"Refresher.FetchInterval",
+		"Update.Method",
+		"Update.Days",
+	}
+
+	changedConfigs := []string{}
+	for _, config := range configsThatDontAutoReload {
+		old := reflect.ValueOf(oldConfig).Elem()
+		new := reflect.ValueOf(newConfig).Elem()
+		fieldNames := strings.Split(config, ".")
+		// navigate to the leaves in old and new config
+		for _, fieldName := range fieldNames {
+			old = old.FieldByName(fieldName)
+			new = new.FieldByName(fieldName)
+		}
+		// if the value has changed, ...
+		if !old.Equal(new) {
+			// ... convert the field names to the user-facing names by
+			// lower-casing the first letter, ...
```

Comment:
This is not guaranteed to work because some fields don't match with their serialized name e.g.:
```go
SuspendOnEdit *bool `yaml:"editInTerminal,omitempty"`
```

Arguably we should just not have mismatched field names, but the reason this happens is that we think of a better name to use internally, but don't want to break backwards compatibility. What are your thoughts on parsing the tag string to pull out the serialized name?

Reply:
I was originally planning to read the struct tags, but then was too lazy to figure out how to do that, and hoped I could get away with this simpler version. But you're right, it's not good enough, and actually reading the struct tags is not so hard, see 292e74c4c31c.